### ISSUE TITLE
[boo]  different loss function for grad numerics testing

### ISSUE THREAD

**Issue by ftynse** [iree-org/iree-turbine#1021]
The naive loss function (sum of half of squared output) might contribute significantly to the errors we see in backward implementations (e.g. even pytorch cpu and gpu numerics differ significantly in many cases).

We could maybe instead directly launch `torch.ops.aten.convolution_backward` on some example inputs instead of also computing the forward pass.

_Originally posted by @zjgarvey in https://github.com/iree-org/iree-turbine/pull/1012#discussion_r2190645126_

An alternative could be to normalize the output first to the [0, 1) range.
